src/mixins/canvas_events.mixin.js

```diff
@@ -715,6 +731,20 @@
 
       if (target) {
         var alreadySelected = target === this._activeObject;
+        if (!alreadySelected) {
+          if (fabric.util.isTouchEvent(e)) {
+            var activeOnDown = (target.setActiveOn === 'mouseuptouchdown' ||
+            target.setActiveOn === 'mousetouchdown');
+          }
+          else {
+            var activeOnDown = (target.setActiveOn === 'mousetouchdown' ||
+            target.setActiveOn === 'mousedowntouchup');
+          }
+          if (!activeOnDown) {
+            this._handleEvent(e, 'down');
+            return;
```

Comment:
early returning like this will make harder to mantain the function. Whatever is after this point we have to be sure that is not needed outside the selection scope.

Reply:
Good point, I'll do this a different way